clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -564,6 +564,21 @@ CodeGenFunction::DecodeAddrUsedInPrologue(llvm::Value *F,
                             "decoded_addr");
 }
 
+void CodeGenFunction::EmitSubGroupMetadata(const FunctionDecl *FD,
+                                           llvm::Function *Fn) {
+  if (const IntelReqdSubGroupSizeAttr *A =
```

Comment:
Can we remove this check and have a check where the function is called instead?

Reply:
I do not think we can do this. We still need to provide attribute information when we have A->getValue()

Reply:
You can still save the Attribute here without the check. I was talking about moving the check to where this is called (in StartFunction), so that this function is only called from `StartFunction` when attribute is present. 

But this is a nit. Functionality remains same with or without this change. You can ignore this if you disagree.

Reply:
I misunderstood your comments. Yes, we can add the check for the attribute.  i have added it with the current patch.